api/openapi/kubeshop-kusk-gateway-api.yaml

```diff
@@ -0,0 +1,210 @@
+  description: ""
+  version: 1.0.0
+servers:
+- url: /
+paths:
+  /apis:
+    get:
+      description: gets a list of APIs
+      operationId: getApis
+      parameters:
+      - name: fleet
+        in: query
+        description: optional filter on fleet
+        required: false
+        style: form
+        explode: true
+        schema:
+          type: string
+      responses:
+        "200":
+          description: a list of apis
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ApiItem'
+  /apis/{apiId}/openapispec:
+    get:
+      description: gets the OpenAPI spec for this api id
+      operationId: getApiOpenApiSpec
+      parameters:
+      - name: apiId
+        in: path
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: string
+          format: uuid
+      responses:
+        "200":
+          description: OpenAPI spec object
+          content:
+            application/json:
+              schema:
+                type: object
+  /services:
+    get:
+      description: list all services
+      operationId: getServices
+      parameters:
+      - name: namespace
+        in: query
+        description: optional filter on namespace
+        required: false
+        style: form
+        explode: true
+        schema:
+          type: string
+      responses:
+        "200":
+          description: list of services
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ServiceItem'
+  /services/{namespace}/{name}:
+    get:
+      description: get details for a single service
+      operationId: getService
+      parameters:
+      - name: namespace
+        in: path
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: string
+      - name: name
+        in: path
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: string
+      responses:
+        "200":
+          description: service details
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ServiceItem'
+  /fleets:
+    get:
+      description: list all envoy fleets
+      operationId: getEnvoyFleets
+      parameters:
+      - name: namespace
+        in: query
+        description: optional filter on namespace
+        required: false
+        style: form
+        explode: true
+        schema:
+          type: string
+      responses:
+        "200":
+          description: list of envoy fleets
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/EnvoyFleetItem'
+  /fleets/{namespace}/{name}:
```

Comment:
as above - could this be /fleets/{fleetId} instead? (fleetId being an internal id)

Reply:
I think /fleets/{id} with ?namespace= filter is the correct way. Otherwise, by the REST API design conventions, we'd need to provide /fleets/{namespace}/, which is meaningless in our case.